submissions/angels_unit_tests/angels_unit_tests.md

```diff
@@ -0,0 +1,620 @@
+
+It took until the 27th of January 2020 to convince Arch666Angel to let the community contribute to his creation. After working out all the details, he selected few people to keep his vision of the mods alive, at which point we could inform the public about this. The main interaction at the time went through the factorio forums, from which you can still read the [announcement message](https://forums.factorio.com/viewtopic.php?p=475786#p475786) till this day.
+
+During the month January 2020 we had already eagerly started to update the mods from 0.17 to 0.18 with all the breaking changes the base game introduced. After Arch666Angel's approval, we released our first version [*"Initial 0.18 release"*](https://github.com/Arch666Angel/mods/commit/8b7ff10ba82d3071f374e08452b5ca22d42e4780)! Soon after celebrating our succes, reality slapped us in the face with a great amount of bug reports on the forums. Little did we know how huge this undertaking would be, how much time we spend bugfixing and most of all, how great the big community is by trying to help figure out the issues.
+
+## It started small
+
+With the great help of the community we learned how the Angel's libary worked. For those that do not know what a library is, it is a set of helper functions that abstract the factorio code base such that it becomes easier to program. It is mainly a list with quality of life additions for the developer, such that he can focus on developing features, rather than writing boiler plate code to get something to work. You can compare Angel's library to some stand-alone mods like the [Factorio Library](https://mods.factorio.com/mod/flib), [Bob's Functions Library mod](https://mods.factorio.com/mod/boblibrary) or even my own attempt [Lovely_santa's library of knowledge](https://mods.factorio.com/mod/LSlib). There is no 'right' or 'wrong' way to implement a library, as long as it suits its purpose. By understanding Angel's library, which actually lives inside Angel's refining and not as a stand-alone mod, we figured more efficient ways of dealing with certain bugs arrising in the mod.
+
+I am confident now saying that we do not make coding mistakes anymore, code slipping through the cracks, which would result in a hard crash of the mods. Nowadays the issues are likely edge cases, unhandled cases or a strange modpack configuration. Most frequently we receive bugs from people who play with a few Angel's mods and a few Bob mods. The reason for this is simple: there are many configurations with many settings, which result in many oversights. [A small example](https://www.reddit.com/r/factorio/comments/pfvqn0/playing_an_angel_bobs_game_and_noticed_about_9/) is shown below; you fix a bug with some technology being unresearchable, in some exotic configuration:
+
+![Nickel processing is unresearchable](https://raw.githubusercontent.com/LovelySanta/Alt-F4/angels-unit-tests/submissions/angels_unit_tests/nickel_processing_bug_1.png)
```

Comment:
These two images is a little unclear as to what we should be looking at. Could you put captions explaining what went wrong?

And if only you have time, the images are very big. You could drag the game window a little smaller and have the information packed together closer. But that's only nit-picking.

Reply:
I've expanded the description in https://github.com/AlternativeFFFF/Alt-F4/pull/551/commits/09e0500a041063c47fe716375139b83f823eb7e9.

I cannot retake these pictures as these are solely pictures taken respectively from reddit and discord as they were bug reports from other people to the angels dev team. Feel free to compress/resize them as needed.